templates/stencil/editor.html.twig

```diff
@@ -0,0 +1,162 @@
+                {% endif %}
+                <div class="col-auto">
+                    <input type="text" id="zone_number" class="form-control" placeholder="">
+                </div>
+            </div>
+        </div>
+        <div class="col-auto">
+            <button type="button" id="save-zone-data" class="btn btn-primary">
+                <i class="ti ti-device-floppy"></i>
+                {% if params.save_zone_data_label is defined %}
+                    <span>{{ params['save_zone_data_label'] }}</span>
+                {% else %}
+                    <span>{{ __("Save zone data") }}</span>
+                {% endif %}
+            </button>
+            <button type="button" id="cancel-zone-data" class="btn btn-outline-danger">
+                <i class="ti ti-x"></i>
+                <span>{{ __("Cancel") }}</span>
+            </button>
+        </div>
+
+        <div class="btn-group col-auto" role="group">
+            <input type="radio" name="editor_mode" class="btn-check" value="crop_editor" id="crop_editor" autocomplete="off" checked>
+            <label class="btn btn-outline-secondary" for="crop_editor">
+                <i class="ti ti-crop"></i>
+                <span>{{ __("Crop") }}</span>
+            </label>
+            <input type="radio" name="editor_mode" class="btn-check" value="move_editor" id="move_editor" autocomplete="off">
+            <label class="btn btn-outline-secondary" for="move_editor">
+                <i class="ti ti-drag-drop"></i>
+                <span>{{ __("Move") }}</span>
+            </label>
+        </div>
+    </div>
+
+    <div class="col-12 mt-2" id="general-submit">
+        {% if id > 0 %}
+            <button type="submit" name="add-new-zone" class="btn btn-primary">
+                <i class="ti ti-plus"></i>
+                <span>{{ params['add_zone_label'] ?? __("Add zone") }}</span>
+            </button>
+            <button type="submit" name="purge" class="btn btn-outline-danger">
+                <i class="ti ti-trash-x"></i>
+                <span>{{ __("Clear data") }}</span>
+            </button>
+        {% else %}
+            <button type="submit" name="add" class="btn btn-primary">
+                <i class="ti ti-plus"></i>
+                <span>{{ __("Add") }}</span>
+            </button>
+        {% endif %}
+    </div>
+</form>
+
+{% for picture in pictures %}
+    <div class="cropper-container mb-2">
+        <img src="{{ picture }}" class="cropper-hidden stencil-image" data-side="{{ loop.index - 1 }}">
+        {% include 'stencil/parts/zones.html.twig' with {
+            'item': item,
+            'stencil': stencil,
+            'zones': zones,
+            'current_side': loop.index - 1,
+            'params': params
+        } only %}
+    </div>
+{% endfor %}
+
+<script type="text/javascript">
+    let editor = new StencilEditor(JSON.parse('{{ zones_json|raw }}'));
```

Comment:
```suggestion
    let editor = new StencilEditor({{ zones_json|json_encode|raw }});
```

Reply:
This modification is not correct. I use `JSON.parse()` to convert a JSON string into a JavaScript object, while `json_encode` generates a JSON string from a PHP array.
I haven't found a cleaner alternative to the original code.

Reply:
My bad, you indeed store a JSON string in db.